src/GraphQL/Resolvers/AsyncEventStreamResolver.cs

```diff
@@ -1,44 +1,54 @@
-using GraphQL.Subscription;
-
 namespace GraphQL.Resolvers
 {
-    public class AsyncEventStreamResolver<T> : IEventStreamResolver<T>
+    public class AsyncEventStreamResolver<T> : IEventStreamResolver
```

Comment:
I would remove `AsyncEventStreamResolver` classes at all and add ctors in `EventStreamResolver` classes. Those are effectively the same concept - resolver returning IObservable.

Reply:
Or even just remove them because `Task<T>`  can be converter by `ValueTask<T>` by async/await trick.

Reply:
I'd like to work on that and `AsyncFieldResolver` in a separate PR.  There are some complications which we will need to consider.

Reply:
> Or even just remove them because `Task<T>` can be converter by `ValueTask<T>` by async/await trick.

Yes that is one option.

Reply:
I would remove indeed. Compiler is smart enough to make things under the hood that we should not write manually complicated the code.

Reply:
I'll do it in a subsequent PR.  I would like to make changes in smaller PRs when possible, and I do not believe anything in this PR requires this change.

Reply:
- #2996